src/sql/workbench/browser/designer/designer.ts

```diff
@@ -941,6 +1031,54 @@ export class Designer extends Disposable implements IThemable {
 		return component;
 	}
 
+	private addDesignerTaskbar(container: HTMLElement, tableProperties: DesignerTableProperties): void {
+		if (tableProperties.canAddRows || tableProperties.canMoveRows) {
+			let taskbarContainer = container.appendChild(DOM.$('.full-row')).appendChild(DOM.$('.add-row-button-container'));
+			this._taskbar = new Taskbar(taskbarContainer);
+			this._actions = [];
+			if (tableProperties.canAddRows) {
+				let addColAction = this._instantiationService.createInstance(AddRowAction, this, tableProperties);
+				this._actions.push(addColAction);
+			}
+			if (tableProperties.canMoveRows) {
+				let moveUpAction = this._instantiationService.createInstance(MoveRowUpAction, this);
+				let moveDownAction = this._instantiationService.createInstance(MoveRowDownAction, this);
+				this._actions.push(moveUpAction);
+				this._actions.push(moveDownAction);
+			}
+			let taskbarContent: ITaskbarContent[] = this._actions.map((a) => { return { action: a }; });
+			this._taskbar.setContent(taskbarContent);
+		}
+	}
+
+	private openContextMenu(table: Table<Slick.SlickData>, event: ITableMouseEvent, propertyPath: DesignerPropertyPath, view: DesignerUIArea): void {
+		const rowIndex = event.cell.row;
+		let designerActionContext: DesignerTableActionContext = {
+			table: table,
+			path: propertyPath,
+			source: view,
+			selectedRow: rowIndex
+		};
+		let data = table.grid.getData() as Slick.DataProvider<Slick.SlickData>;
+		if (!data || rowIndex >= data.getLength()) {
+			return undefined;
+		}
+		let actions = this.getDesignerActions();
+		this._contextMenuService.showContextMenu({
+			getAnchor: () => event.anchor,
+			getActions: () => actions,
+			getActionsContext: () => (designerActionContext)
+		});
+	}
+
+	private getDesignerActions(): IAction[] {
+		let actions: IAction[];
+		let insertRowBefore = this._instantiationService.createInstance(InsertBeforeSelectedRowAction, this);
```

Comment:
the actions are hard coded here, what if I want to add move up/down actions. please refactor the code to make it possible.

Reply:
as I mentioned in another comment, this is not designer actions, this is table level actions, you should decide which actions to add by checking the table's definition.

Reply:
I think a refactoring is needed, we need to add move up/down to context menu for accessibility.

Reply:
this wasn't in the initial task.
do you want me to add the move actions in the context menu?

Reply:
yes, we still have time, please add the actions.